frontend/src/components/pages/WaitlistExperience/index.tsx

```diff
@@ -1,15 +1,83 @@
-import React, { useState } from "react";
-import WaitlistConfirmationPage from "./ConfirmationPage";
+import { Center, Spinner } from "@chakra-ui/react";
+import React, { useEffect, useRef, useState } from "react";
+import { useParams } from "react-router-dom";
+import CampsAPIClient from "../../../APIClients/CampsAPIClient";
+import { WaitListExperienceCamper } from "../../../types/CamperTypes";
+import { CampResponse, CampSession } from "../../../types/CampsTypes";
+import WaitlistConfirmationPage from "./ConfirmationPage/ConfirmationPage";
 import WaitlistPersonalInfoPage from "./PersonalInfo";
+import { checkPersonalInfoFilled } from "./PersonalInfo/waitListInfoReducer";
 import WaitlistFooter from "./WaitlistFooter";
 
-const WaitlistExperiencePage = (): React.ReactElement => {
-  const [isConfirmationStep, setIsConfirmationStep] = useState(false);
+interface WaitlistExperiencePageProps {
+  campSessions: CampSession[];
+}
+
+const WaitlistExperiencePage = ({
+  campSessions,
+}: WaitlistExperiencePageProps): React.ReactElement => {
+  const { id: campId } = useParams<{ id: string }>();
+  const [camp, setCamp] = useState<CampResponse | undefined>(undefined);
+  const submitBtnRef = useRef<HTMLButtonElement>(null);
+  const [campers, setCampers] = useState<WaitListExperienceCamper[]>([
+    {
+      firstName: "",
+      lastName: "",
+      age: NaN,
+      contact: {
+        firstName: "",
+        lastName: "",
+        email: "",
+        phoneNumber: "",
+        relationshipToCamper: "",
+      }, // Deep copy the contact of the 1st camper's contacts as single source of truth
+    },
+  ]);
+  useEffect(() => {
+    CampsAPIClient.getCampById(campId).then((campResponse) => {
+      if (campResponse.id) {
+        setCamp(campResponse);
+      }
+    });
+  }, [campId]);
 
+  const [isConfirmationStep, setIsConfirmationStep] = useState(false);
   // Temp variables for personal info page
-  const [isPersonalInfoChecked, setIsPersonalInfoChecked] = useState(false);
+  const isPersonalInfoComplete = checkPersonalInfoFilled(campers, camp);
+
+  if (!camp)
+    return (
+      <Center>
+        <Spinner />
+      </Center>
+    );
 
-  const isPersonalInfoComplete = isPersonalInfoChecked;
+  if (!campSessions) {
```

Comment:
I don't think we need this. selectedCampSessions should always be a non empty array of campSessions and we should never hard code campSession data

Reply:
Yep, shall be removed